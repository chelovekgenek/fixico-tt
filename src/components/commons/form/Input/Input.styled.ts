import styled from "styled-components";

import TextField from "@material-ui/core/TextField";

export const InputWrapper = styled(TextField)`
  width: 100%;
`;
