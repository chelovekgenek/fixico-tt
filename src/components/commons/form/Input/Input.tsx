import React from "react";
import { Field, FieldProps } from "formik";

import { InputWrapper } from "./Input.styled";

interface IProps {
  name: string;
  label: string;
}

export const Input: React.FC<IProps> = ({ name, label, ...props }) => (
  <Field name={name}>
    {(props: FieldProps) => (
      <InnerFormItem name={name} label={label} {...props} />
    )}
  </Field>
);

export const InnerFormItem: React.FC<IProps & FieldProps> = ({
  label,
  name,
  field,
  form,
  ...props
}) => (
  <InputWrapper
    label={label}
    id={name}
    error={Object.keys(form.errors).includes(name)}
    helperText={form.errors[name]}
    value={field.value || ""}
    onChange={field.onChange}
    onBlur={field.onBlur}
    {...props}
  />
);
