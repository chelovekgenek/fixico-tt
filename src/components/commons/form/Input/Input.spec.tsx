import React from "react";
import { shallow } from "enzyme";

import fieldProps from "mocks/formik-field-props";
import formProps from "mocks/formik-form-props";

import { InnerFormItem } from "./Input";

describe("Input", () => {
  const defaultProps = {
    ...fieldProps,
    field: { ...fieldProps.field, value: "value", name: "input" },
    name: "input",
    label: "input",
    value: "value"
  };
  const render = (
    props: Partial<React.ComponentProps<typeof InnerFormItem>> = {}
  ) => shallow(<InnerFormItem {...defaultProps} {...props} />);

  it("should render", () => {
    expect(render()).toMatchSnapshot();
  });
  it("should render errors", () => {
    expect(
      render({ form: { ...formProps, errors: { input: "error" } } })
    ).toMatchSnapshot();
  });
});
