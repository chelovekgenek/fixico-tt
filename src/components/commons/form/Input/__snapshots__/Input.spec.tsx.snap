// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<Styled(WithStyles(ForwardRef(TextField)))
  error={false}
  id="input"
  label="input"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  value="value"
/>
`;

exports[`Input should render errors 1`] = `
<Styled(WithStyles(ForwardRef(TextField)))
  error={true}
  helperText="error"
  id="input"
  label="input"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  value="value"
/>
`;
